src/gleam/result.gleam

```diff
@@ -366,6 +366,51 @@ pub fn all(results: List(Result(a, e))) -> Result(List(a), e) {
   list.try_map(results, fn(x) { x })
 }
 
+/// Combines a list of results into a single result.
+/// If all elements in the list are `Ok` then returns an `Ok` holding the list of values.
+/// If any element is `Error` then returns an `Error` holding the list of all errors.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > partition([Ok(1), Ok(2)])
+/// Ok([1, 2])
+/// ```
+///
+/// ```gleam
+/// > partition([Ok(1), Error("a"), Error("b")])
+/// Error(["a", "b"])
+/// ```
+///
+pub fn partition(results: List(Result(a, e))) -> Result(List(a), List(e)) {
```

Comment:
I think partition would be `fn(List(Result(a, e)) -> List(#(a, e))`, to match `list.partition`.

https://hexdocs.pm/gleam_stdlib/gleam/list.html#partition

Reply:
Sorry, yes, typo!